src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,200 @@
+            first.setPrev(addedNode);
+            first = addedNode;
+            size++;
+        } else {
+            Node searchedNode = findNodeByIndex(index);
+            Node addedNode = new Node(value);
+            addedNode.setNext(searchedNode);
+            addedNode.setPrev(searchedNode.getPrev());
+            searchedNode.getPrev().setNext(addedNode);
+            searchedNode.setPrev(addedNode);
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T node : list) {
+            add(node);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node searchedNode = findNodeByIndex(index);
+        return searchedNode.getValue();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node node = findNodeByIndex(index);
+        T oldValue = node.getValue();
+        node.setValue(value);
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node node = findNodeByIndex(index);
+        if (node == first) {
+            if (first == last) {
+                first = null;
+                last = null;
+            } else {
+                first = node.getNext();
+                node.getNext().setPrev(null);
+            }
+        } else if (node == last) {
+            last = node.getPrev();
+            node.getPrev().setNext(null);
+        } else {
+            node.getPrev().setNext(node.getNext());
+            node.getNext().setPrev(node.getPrev());
+        }
+        size--;
+        return node.getValue();
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node node = first;
+        int index = 0;
+        while (node != null && !Objects.equals(node.getValue(), object) && index != size) {
+            node = node.getNext();
+            index++;
+        }
+        if (node == null) {
+            return false;
+        }
+        remove(index);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public Node getFirst() {
+        return first;
+    }
+
+    public Node getLast() {
+        return last;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public void setFirst(Node first) {
+        this.first = first;
+    }
+
+    public void setLast(Node last) {
+        this.last = last;
+    }
+
+    public void setSize(int size) {
+        this.size = size;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index is out of bound");
+        }
+    }
+
+    private Node findNodeByIndex(int index) {
```

Comment:
You have several mistakes from the common mistakes checklist, please review and fix them. Thanks!
(one of them)

![image](https://github.com/mate-academy/jv-linked-list/assets/72043323/db52ce42-3e98-42c2-be9b-399d44744696)